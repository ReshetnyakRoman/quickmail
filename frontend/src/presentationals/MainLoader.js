import React from 'react'
import connectedToSocials from '../containers/connectedToSocials'
import withLoginToSocials from '../containers/withLoginToSocials'

function MainLoader (props){
  return (
    <div className='main-loader flex-column'>
      <div className='loader-container '>
        <div className='envelope'>
          <div className='envelope-left' />
          <div className='envelope-right' />
        </div>
        <div className='envelope-content' />
        <div className='envelope-top' />
      </div>
      <div className='progress' style={{height: '2px'}}>
        <div className='progress-bar'
          role='progressbar'
          aria-valuenow='100'
          aria-valuemin='0'
          aria-valuemax='100' />
      </div>
    </div>
  )
}

const MainLoadeWithLoginCheck = withLoginToSocials(connectedToSocials(MainLoader))
export default MainLoadeWithLoginCheck
