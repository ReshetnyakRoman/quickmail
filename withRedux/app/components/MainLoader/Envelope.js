import styled, { keyframes } from 'styled-components';

const open = keyframes`
  0%, 20% {
    opacity: 0;
    top: 16px;
  }
  20%, 50% {
    opacity: 0;
  }
  55%, 100% {
    opacity: 1;
    top: 155px;
  }
`;

const flip = keyframes`
  0%, 60% {
    border-width: 0 144px 110px 144px;
    -webkit-transform: rotate3d(1, 0, 0, 180deg);
    -moz-transform: rotate3d(1, 0, 0, 180deg);
    -o-transform: rotate3d(1, 0, 0, 180deg);
    -ms-transform: rotate3d(1, 0, 0, 180deg);
    transform: rotate3d(1, 0, 0, 180deg);
  }
  60%, 100% {
    border-width: 0 144px 110px 144px;
    -webkit-transform: rotate3d(1, 0, 0, 0deg);
    -moz-transform: rotate3d(1, 0, 0, 0deg);
    -o-transform: rotate3d(1, 0, 0, 0deg);
    -ms-transform: rotate3d(1, 0, 0, 0deg);
    transform: rotate3d(1, 0, 0, 0deg);
  }
`;

const openSmall = keyframes`
  0%, 20% {
    opacity: 0;
    -ms-filter: "progid:DXImageTransform.Microsoft.Alpha(Opacity=0)";
    filter: alpha(opacity=0);
    top: 160px;
  }
  20%, 50% {
    opacity: 0;
    -ms-filter: "progid:DXImageTransform.Microsoft.Alpha(Opacity=0)";
    filter: alpha(opacity=0);
  }
  55%, 100% {
    opacity: 1;
    -ms-filter: none;
    filter: none;
    top: 90px;
  }
`;

const flipSmall = keyframes`
  0%, 60% {
    border-width: 0 72px 55px 72px;
    -webkit-transform: rotate3d(1, 0, 0, 180deg);
    -moz-transform: rotate3d(1, 0, 0, 180deg);
    -o-transform: rotate3d(1, 0, 0, 180deg);
    -ms-transform: rotate3d(1, 0, 0, 180deg);
    transform: rotate3d(1, 0, 0, 180deg);
  }
  60%, 100% {
    border-width: 0 70px 55px 70px;
    -webkit-transform: rotate3d(1, 0, 0, 0deg);
    -moz-transform: rotate3d(1, 0, 0, 0deg);
    -o-transform: rotate3d(1, 0, 0, 0deg);
    -ms-transform: rotate3d(1, 0, 0, 0deg);
    transform: rotate3d(1, 0, 0, 0deg);
  }
`;

const Envelope = styled.div`
  background-color: #3267d5;
  width: 300px;
  height: 180px;
  margin: 0 auto 100px auto;
  border-top: 0px solid #ececec;
  border-radius: 20px;
  position: relative;
  overflow: hidden;
  @media (max-width: 768px) {
    width: 150px;
    height: 90px;
    margin: 0 auto 20px auto;
    border-top: 2px solid #ececec;
    border-radius: 10px;
  }
`;

const EnvelopeLeft = styled.div`
  width: 0;
  height: 0;
  border-style: solid;
  border-width: 0 270px 170px 0;
  border-color: transparent transparent #2ab6f6 transparent;
  position: absolute;
  left: 0;
  bottom: 0;
  @media (max-width: 768px) {
    border-width: 0 135px 85px 0;
  }
`;

const EnvelopeRight = styled.div`
  width: 0;
  height: 0;
  border-style: solid;
  border-width: 0 0 170px 270px;
  border-color: transparent transparent #81d2f9 transparent;
  position: absolute;
  right: 0;
  bottom: 0;
  @media (max-width: 768px) {
    border-width: 0 0 85px 135px;
  }
`;

const EnvelopeTop = styled.div`
  width: 0px;
  height: 0;
  border-style: solid;
  border-color: transparent transparent #2a55c5 transparent;
  position: absolute;
  top: 16px;
  -webkit-animation: ${flip} 3s linear infinite;
  -moz-animation: ${flip} 3s linear infinite;
  -o-animation: ${flip} 3s linear infinite;
  -ms-animation: ${flip} 3s linear infinite;
  animation: ${flip} 3s linear infinite;
  -webkit-perspective: 800px;
  -moz-perspective: 800px;
  -ms-perspective: 800px;
  perspective: 800px;
  -webkit-transform-origin: 100% 100%;
  -moz-transform-origin: 100% 100%;
  -o-transform-origin: 100% 100%;
  -ms-transform-origin: 100% 100%;
  transform-origin: 100% 100%;
  @media (max-width: 768px) {
    top: 38px;
    left: 29px;
    -webkit-animation: ${flipSmall} 3s linear infinite;
    -moz-animation: ${flipSmall} 3s linear infinite;
    -o-animation: ${flipSmall} 3s linear infinite;
    -ms-animation: ${flipSmall} 3s linear infinite;
    animation: ${flipSmall} 3s linear infinite;
  }
`;

const EnvelopeContent = styled.div`
  width: 70px;
  height: 20px;
  background-color: #e0e0e0;
  -webkit-transform: rotate(45deg);
  -moz-transform: rotate(45deg);
  -o-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
  position: absolute;
  left: 140px;
  top: 280px;
  z-index: 11;
  -webkit-animation: ${open} 3s linear infinite;
  -moz-animation: ${open} 3s linear infinite;
  -o-animation: ${open} 3s linear infinite;
  -ms-animation: ${open} 3s linear infinite;
  animation: ${open} 3s linear infinite;
  opacity: 0;
  &::after {
    content: '';
    width: 200%;
    height: 100%;
    background-color: #e0e0e0;
    position: absolute;
    left: -10px;
    bottom: 60px;
    -webkit-transform: rotate(-90deg);
    -moz-transform: rotate(-90deg);
    -o-transform: rotate(-90deg);
    -ms-transform: rotate(-90deg);
    transform: rotate(-90deg);
  }
  @media (max-width: 768px) {
    width: 35px;
    height: 10px;
    left: 75px;
    top: 90px;
    -webkit-animation: ${openSmall} 3s linear infinite;
    -moz-animation: ${openSmall} 3s linear infinite;
    -o-animation: ${openSmall} 3s linear infinite;
    -ms-animation: ${openSmall} 3s linear infinite;
    animation: ${openSmall} 3s linear infinite;
    &::after {
      left: -5px;
      bottom: 30px;
    }
  }
`;

export default Envelope;
export { EnvelopeLeft, EnvelopeRight, EnvelopeTop, EnvelopeContent };
