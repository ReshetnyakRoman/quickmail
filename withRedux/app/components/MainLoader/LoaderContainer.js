import styled from 'styled-components';

const LoaderContainer = styled.div`
  width: 400px;
  height: 400px;
  border-radius: 20%;
  display: -webkit-flex;
  display: flex;
  -webkit-box-pack: center;
  -moz-box-pack: center;
  -o-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-box-align: end;
  -moz-box-align: end;
  -o-box-align: end;
  -ms-flex-align: end;
  -webkit-align-items: flex-end;
  align-items: flex-end;
  position: relative;
  @media (max-width: 768px) {
    width: 200px;
    height: 200px;
    border-radius: 20%;
  }
`;

export default LoaderContainer;
