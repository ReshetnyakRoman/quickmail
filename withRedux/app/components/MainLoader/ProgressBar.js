import styled from 'styled-components';
import LinearProgress from '@material-ui/core/LinearProgress';

const ProgressBar = styled(LinearProgress)`
  width: 100%;
  height: 2px !important;
  & div {
    background-color: #3267d5;
  }
`;

export default ProgressBar;
