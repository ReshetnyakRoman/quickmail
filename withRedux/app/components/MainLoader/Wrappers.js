import styled from 'styled-components';

const Wrapper = styled.div`
  position: fixed;
  -webkit-box-align: center;
  -moz-box-align: center;
  -o-box-align: center;
  -ms-flex-align: center;
  -webkit-align-items: center;
  align-items: center;
  -webkit-box-pack: center;
  -moz-box-pack: center;
  -o-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  width: 100%;
  height: 100%;
  top: 0%;
  left: 0%;
  background-color: #fff;
  z-index: 10;
  display: -webkit-flex;
  display: flex;
  flex-direction: column;
`;

const ProgressWrapper = styled.div`
  width: 70%;
`;

export { Wrapper, ProgressWrapper };
