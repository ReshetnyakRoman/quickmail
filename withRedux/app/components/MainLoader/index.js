/*
 * This is main loader which user see while app loading
 * and getting authorization info from VK and FB socials
 * to define logged in status
 * if logged in it'll redirect to /inbox other wise to /welcome
*/

import React from 'react';
import withLoginToSocials from 'containers/LoginWithSocials/WithLoginToSocials';
import connectToSocials from 'containers/LoginWithSocials/ConnectToSocials';
import ProgressBar from './ProgressBar';
import { Wrapper, ProgressWrapper } from './Wrappers';
import LoaderContainer from './LoaderContainer';
import Envelope, {
  EnvelopeLeft,
  EnvelopeRight,
  EnvelopeTop,
  EnvelopeContent,
} from './Envelope';

const MainLoader = () => (
  <Wrapper>
    <LoaderContainer>
      <Envelope>
        <EnvelopeLeft />
        <EnvelopeRight />
      </Envelope>
      <EnvelopeContent />
      <EnvelopeTop />
    </LoaderContainer>
    <ProgressWrapper>
      <ProgressBar />
    </ProgressWrapper>
  </Wrapper>
);

export { MainLoader };

const MainLoadeWithLoginCheck = withLoginToSocials(
  connectToSocials(MainLoader),
);
export default MainLoadeWithLoginCheck;
