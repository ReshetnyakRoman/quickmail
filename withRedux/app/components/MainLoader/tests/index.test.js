import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import MainLoader from '../index';

describe('<MainLoader />', () => {
  it('should match the snapshot', () => {
    const rendredComponent = renderer.create(<MainLoader />).toJSON();
    expect(rendredComponent).toMatchSnapshot();
  });
});
