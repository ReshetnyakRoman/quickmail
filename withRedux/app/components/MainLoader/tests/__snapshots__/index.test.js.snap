// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainLoader /> should match the snapshot 1`] = `
.c0 {
  position: fixed;
  -webkit-box-align: center;
  -moz-box-align: center;
  -o-box-align: center;
  -ms-flex-align: center;
  -webkit-align-items: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -moz-box-pack: center;
  -o-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  height: 100%;
  top: 0%;
  left: 0%;
  background-color: #fff;
  z-index: 10;
  display: -webkit-flex;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  width: 400px;
  height: 400px;
  border-radius: 20%;
  display: -webkit-flex;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -moz-box-pack: center;
  -o-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-box-align: end;
  -moz-box-align: end;
  -o-box-align: end;
  -ms-flex-align: end;
  -webkit-align-items: flex-end;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  position: relative;
}

.c2 {
  background-color: #3267d5;
  width: 300px;
  height: 180px;
  margin: 0 auto 100px auto;
  border-top: 0px solid #ececec;
  border-radius: 20px;
  position: relative;
  overflow: hidden;
}

.c3 {
  width: 0;
  height: 0;
  border-style: solid;
  border-width: 0 270px 170px 0;
  border-color: transparent transparent #2ab6f6 transparent;
  position: absolute;
  left: 0;
  bottom: 0;
}

.c4 {
  width: 0;
  height: 0;
  border-style: solid;
  border-width: 0 0 170px 270px;
  border-color: transparent transparent #81d2f9 transparent;
  position: absolute;
  right: 0;
  bottom: 0;
}

.c6 {
  width: 0px;
  height: 0;
  border-style: solid;
  border-color: transparent transparent #2a55c5 transparent;
  position: absolute;
  top: 16px;
  -webkit-animation: jPiUVz 3s linear infinite;
  -moz-animation: jPiUVz 3s linear infinite;
  -o-animation: jPiUVz 3s linear infinite;
  -ms-animation: jPiUVz 3s linear infinite;
  -webkit-animation: jPiUVz 3s linear infinite;
  animation: jPiUVz 3s linear infinite;
  -webkit-perspective: 800px;
  -moz-perspective: 800px;
  -ms-perspective: 800px;
  -webkit-perspective: 800px;
  -moz-perspective: 800px;
  -ms-perspective: 800px;
  perspective: 800px;
  -webkit-transform-origin: 100% 100%;
  -moz-transform-origin: 100% 100%;
  -o-transform-origin: 100% 100%;
  -ms-transform-origin: 100% 100%;
  -webkit-transform-origin: 100% 100%;
  -ms-transform-origin: 100% 100%;
  transform-origin: 100% 100%;
}

.c5 {
  width: 70px;
  height: 20px;
  background-color: #e0e0e0;
  -webkit-transform: rotate(45deg);
  -moz-transform: rotate(45deg);
  -o-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
  position: absolute;
  left: 140px;
  top: 280px;
  z-index: 11;
  -webkit-animation: eGafGf 3s linear infinite;
  -moz-animation: eGafGf 3s linear infinite;
  -o-animation: eGafGf 3s linear infinite;
  -ms-animation: eGafGf 3s linear infinite;
  -webkit-animation: eGafGf 3s linear infinite;
  animation: eGafGf 3s linear infinite;
  opacity: 0;
}

.c5::after {
  content: '';
  width: 200%;
  height: 100%;
  background-color: #e0e0e0;
  position: absolute;
  left: -10px;
  bottom: 60px;
  -webkit-transform: rotate(-90deg);
  -moz-transform: rotate(-90deg);
  -o-transform: rotate(-90deg);
  -ms-transform: rotate(-90deg);
  -webkit-transform: rotate(-90deg);
  -ms-transform: rotate(-90deg);
  transform: rotate(-90deg);
}

.c7 {
  width: 70%;
  height: 2px !important;
}

.c7 div {
  background-color: #3267d5;
}

@media (max-width:768px) {
  .c1 {
    width: 200px;
    height: 200px;
    border-radius: 20%;
  }
}

@media (max-width:768px) {
  .c2 {
    width: 150px;
    height: 90px;
    margin: 0 auto 20px auto;
    border-top: 2px solid #ececec;
    border-radius: 10px;
  }
}

@media (max-width:768px) {
  .c3 {
    border-width: 0 135px 85px 0;
  }
}

@media (max-width:768px) {
  .c4 {
    border-width: 0 0 85px 135px;
  }
}

@media (max-width:768px) {
  .c6 {
    top: 38px;
    left: 29px;
    -webkit-animation: hgRlbf 3s linear infinite;
    -moz-animation: hgRlbf 3s linear infinite;
    -o-animation: hgRlbf 3s linear infinite;
    -ms-animation: hgRlbf 3s linear infinite;
    -webkit-animation: hgRlbf 3s linear infinite;
    animation: hgRlbf 3s linear infinite;
  }
}

@media (max-width:768px) {
  .c5 {
    width: 35px;
    height: 10px;
    left: 75px;
    top: 90px;
    -webkit-animation: HQWtC 3s linear infinite;
    -moz-animation: HQWtC 3s linear infinite;
    -o-animation: HQWtC 3s linear infinite;
    -ms-animation: HQWtC 3s linear infinite;
    -webkit-animation: HQWtC 3s linear infinite;
    animation: HQWtC 3s linear infinite;
  }

  .c5::after {
    left: -5px;
    bottom: 30px;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <div
        className="c3"
      />
      <div
        className="c4"
      />
    </div>
    <div
      className="c5"
    />
    <div
      className="c6"
    />
  </div>
  <div
    className="MuiLinearProgress-root-1 MuiLinearProgress-colorPrimary-2 c7"
    role="progressbar"
  >
    <div
      className="MuiLinearProgress-bar-9 MuiLinearProgress-barColorPrimary-10 MuiLinearProgress-bar1Indeterminate-12"
      style={Object {}}
    />
    <div
      className="MuiLinearProgress-bar-9 MuiLinearProgress-barColorPrimary-10 MuiLinearProgress-bar2Indeterminate-15"
      style={Object {}}
    />
  </div>
</div>
`;


